eng/pipelines/swagger-api-doc-preview.yml

```
@@ -0,0 +1,203 @@
+        parameters:
+          State: pending
+          TargetUrl: $(CurrentBuildUrl)
+          Description: 'Starting docs build'
+          Context: $(StatusName)
+
+      - checkout: self
+        # Fetch depth required to get list of changed files
+        fetchDepth: 2
+
+      - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+        parameters:
+          SkipCheckoutNone: true
+          TokenToUseForAuth: $(azuresdk-github-pat)
+          # Path does not need to be set because sparse-checkout.yml already
+          # checks out files in the repo root
+          Repositories:
+            - Name: MicrosoftDocs/AzureRestPreview
+              # TODO: use main branch when this is ready to go live
+              Commitish: refs/heads/doc-preview-migration-test
+              WorkingDirectory: AzureRestPreview
+
+      - template: /eng/pipelines/templates/steps/npm-install.yml
+
+      - script: npm exec --no -- api-doc-preview --output ../../AzureRestPreview
+        displayName: Generate Swagger API documentation preview
+        workingDirectory: .github/shared
+
+      - template: /eng/common/pipelines/templates/steps/git-push-changes.yml
+        parameters:
+          WorkingDirectory: AzureRestPreview
+          BaseRepoOwner: MicrosoftDocs
+          TargetRepoOwner: MicrosoftDocs
+          TargetRepoName: AzureRestPreview
+          BaseRepoBranch: $(BranchName)
+          CommitMsg: |
+            Update API Doc Preview
+            Build: $(System.CollectionUri)$(System.TeamProject)/_build/results?buildId=$(Build.BuildId)
+            PR: $(System.PullRequest.SourceRepositoryURI)/pull/$(System.PullRequest.PullRequestId)
+
+      - task: AzureCLI@2
+        displayName: Start docs build
+        inputs:
+          azureSubscription: msdocs-apidrop-connection
+          scriptType: pscore
+          scriptLocation: inlineScript
+          inlineScript: |
+            $buildStartRaw = az pipelines build queue `
```

could this fail? if so, should we handle error or just let it bubble up?